compiler/damlc/tests/src/DA/Test/Packaging.hs

```diff
@@ -877,6 +877,90 @@ dataDependencyTests damlc repl davlDar oldProjDar = testGroup "Data Dependencies
 
           step "Validating DAR"
           callProcessSilent repl ["validate", tmpDir </> "b" </> "b.dar"]
+
+    , testCaseSteps "Colliding package names" $ \step -> withTempDir $ \tmpDir -> do
+          forM_ ["1", "2"] $ \version -> do
+              step ("Building 'lib" <> version <> "'")
+              let projDir = tmpDir </> "lib-" <> version
+              createDirectoryIfMissing True projDir
+              writeFileUTF8 (projDir </> "daml.yaml") $ unlines
+                  [ "sdk-version: " <> sdkVersion
+                  , "version: " <> show version
+                  , "name: lib"
+                  , "source: ."
+                  , "dependencies: [daml-prim, daml-stdlib]"
+                  ]
+              writeFileUTF8 (projDir </> "Lib.daml") $ unlines
+                  [ "daml 1.2 module Lib where"
+                  , "data X" <> version <> " = X"
+                  ]
+              withCurrentDirectory projDir $ callProcessSilent damlc ["build", "-o", projDir </> "lib.dar"]
+
+          step "Building a"
+          let projDir = tmpDir </> "a"
+          createDirectoryIfMissing True projDir
+          writeFileUTF8 (projDir </> "daml.yaml") $ unlines
+               [ "sdk-version: " <> sdkVersion
+               , "version: 0.0.0"
+               , "name: a"
+               , "source: ."
+               , "dependencies: [daml-prim, daml-stdlib]"
+               , "data-dependencies:"
+               , "- " <> show (tmpDir </> "lib-1" </> "lib.dar")
+               ]
+          writeFileUTF8 (projDir </> "A.daml") $ unlines
+              [ "daml 1.2 module A where"
+              , "import Lib"
+              , "data A = A X1"
+              ]
+          withCurrentDirectory projDir $ callProcessSilent damlc ["build", "-o", projDir </> "a.dar"]
+
+          step "Building b"
```

Comment:
We don’t strictly need both `a` and `b` here if we use `package`-flags to rename `Lib` we could directly reference both from `a`. However, this seems a bit more natural as something that users might run into themselves so I prefer to keep it like this.